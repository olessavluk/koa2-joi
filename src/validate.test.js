'use strict';

const request = require('supertest');
const Koa = require('koa');
const bodyParser = require('koa-bodyparser');
const joi = require('joi');
const util = require('util');
const validate = require('./validate');

let app;
let listener;

beforeEach(() => {
  app = new Koa();

  app
    .use(bodyParser())
    .use(async (ctx, next) => {
      try {
        await next();
      } catch (err) {
        if (err.isJoi) {
          ctx.body = err.details;
          // ctx.app.emit('error', err, ctx);
        } else {
          throw err;
        }
      }
    })
    .use(
      validate({
        body: joi.object().keys({
          test: joi.number().required(),
        }),
      }),
    )
    .use(ctx => {
      ctx.body = ctx.request.body;
    });

  listener = app.listen();
});

afterEach(() => {
  listener.close();
});

it('should reject an invalid request', async () => {
  expect.assertions(1);
  const r = await request(listener)
    .post('/')
    .send({ fail: 'fail' });

  expect(r.body).toMatchSnapshot();
});

it('should accept a valid request', async () => {
  expect.assertions(1);
  const r = await request(listener)
    .post('/')
    .send({ test: 1 });
  expect(r.body).toMatchSnapshot();
});
