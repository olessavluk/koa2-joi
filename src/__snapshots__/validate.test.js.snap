// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept a valid request 1`] = `
Object {
  "test": 1,
}
`;

exports[`should reject an invalid request 1`] = `
Array [
  Object {
    "context": Object {
      "key": "test",
      "label": "test",
    },
    "message": "\\"test\\" is required",
    "path": Array [
      "test",
    ],
    "type": "any.required",
  },
]
`;
